common/src/main/java/org/conscrypt/AbstractSessionContext.java

```diff
@@ -214,6 +214,23 @@ final void cacheSession(NativeSslSession session) {
         }
     }
 
+    /**
+     * Removes the given session from the cache.
```

Comment:
This is keyed by session ID, which means it's also only usable by servers, right? The "Called for server sessions only" comment should probably be here and on `cacheSession` too. (Why are they on the shared class and not the server-specific one?)